crates/router/src/scripts/pii_encryption.rs

```diff
@@ -0,0 +1,373 @@
+        _,
+    >(
+        &conn,
+        ma_dsl::merchant_id.eq(ma_dsl::merchant_id),
+        None,
+        None,
+        None,
+    )
+    .await
+    .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
+    for mer in merchants.iter() {
+        let key = services::generate_aes256_key()
+            .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
+        crate_merchant_key_store(state, &mer.merchant_id, key.to_vec()).await?;
+    }
+    let mut domain_merchants = Vec::with_capacity(merchants.len());
+    for mf in merchants.into_iter() {
+        let merchant_id = mf.merchant_id.clone();
+        let domain_merchant: domain::MerchantAccount = mf
+            .convert(state, &merchant_id, state.get_migration_timestamp())
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)?;
+        domain_merchants.push(domain_merchant);
+    }
+    for m in domain_merchants {
+        let merchant_id = m.merchant_id.clone();
+        let updated_merchant_account = storage::MerchantAccountUpdate::Update {
+            merchant_name: m.merchant_name.clone(),
+            merchant_details: m.merchant_details.clone(),
+            return_url: None,
+            webhook_details: None,
+            sub_merchants_enabled: None,
+            parent_merchant_id: None,
+            api_key: m.api_key.clone(),
+            primary_business_details: None,
+            enable_payment_response_hash: None,
+            payment_response_hash_key: None,
+            redirect_to_merchant_with_http_post: None,
+            routing_algorithm: None,
+            locker_id: None,
+            publishable_key: None,
+            metadata: None,
+            intent_fulfillment_time: None,
+        };
+        crate::logger::warn!("Started for {}", merchant_id);
+
+        conn.transaction_async::<MerchantAccount, async_bb8_diesel::ConnectionError, _, _>(
+            |conn| async move {
+                Conversion::convert(m)
+                    .await
+                    .map_err(|_| {
+                        async_bb8_diesel::ConnectionError::Query(
+                            diesel::result::Error::QueryBuilderError(
+                                "Error while decrypting MerchantAccount".into(),
+                            ),
+                        )
+                    })?
+                    .update(&conn, updated_merchant_account.into())
+                    .await
+                    .map_err(|_| {
+                        async_bb8_diesel::ConnectionError::Query(
+                            diesel::result::Error::QueryBuilderError(
+                                "Error while updating MerchantAccount".into(),
+                            ),
+                        )
+                    })
+            },
+        )
+        .await
+        .into_report()
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
```

Comment:
I only see a single DB operation here. Why does this need to be in a transaction?